addons/survey/controllers/survey_session_manage.py

```diff
@@ -101,18 +96,45 @@ def survey_session_ranking(self, survey_token, **kwargs):
             # no open session
             return ''
 
-        return request.env.ref('survey.user_input_session_ranking').render({
+        return request.env.ref('survey.user_input_session_leaderboard').render({
             'animate_width': True,
-            'ranking': survey._prepare_ranking_values()
+            'leaderboard': survey._prepare_leaderboard_values()
         })
 
     def _prepare_manage_session_values(self, survey):
         is_last_question = False
         if survey.question_ids:
             is_last_question = survey.session_question_id == survey.question_ids[-1]
 
-        return {
+        values = {
             'survey': survey,
             'is_last_question': is_last_question,
             'survey_url': url_join(survey.get_base_url(), survey.get_start_short_url()),
         }
+
+        values.update(self._prepare_question_results_values(survey, request.env['survey.user_input.line']))
+
+        return values
+
+    def _prepare_question_results_values(self, survey, user_input_lines):
+        question = survey.session_question_id
+        question_answers = False
+        if (any(answer.is_correct for answer in question.suggested_answer_ids)):
+            question_answers = [answer.is_correct for answer in question.suggested_answer_ids]
+            if question.comment_count_as_answer:
+                question_answers.append(False)
+
+        full_statistics = question._prepare_statistics(user_input_lines)[0]
+        user_input_lines = None
+        if question.question_type in ['char_box', 'date', 'datetime']:
+            user_input_lines = [{
+                'id': line.id,
+                'value': line['value_%s' % question.question_type]
+            } for line in full_statistics.get('table_data', request.env['survey.user_input.line'])[:100]]
+
+        return {
+            'question_statistics': full_statistics.get('graph_data'),
```

Comment:
I had to browse a bit to understand that question_statistics is not the full results of Question._prepare_statistics but only its graph_data. Maybe we could rename to question_statistics_graph ? It is a bit longer, but we know it comes from _prepare_statistics and limited to its graph_data entry. Indeed I was wondering a bit how you managed the structured answer.